markdown.dtx

```diff
@@ -21956,31 +21975,33 @@ function M.reader.new(writer, options)
 
   parsers.link_image_opening  = Ct( Cg(Cc("delim"), "type")
                                   * Cg(Cc(true), "is_opening")
-                                  * ( Cg(Cc("image"), "element") 
+                                  * ( Cg(Cc("image"), "element")
                                     * parsers.image_opening
                                     * Cg(parsers.exclamation * parsers.lbracket, "content")
                                     + Cg(Cc("link"), "element")
                                     * parsers.link_opening
                                     * Cg(parsers.lbracket, "content")))
 
   parsers.link_image_closing  = Ct( Cg(Cc("delim"), "type")
-                                  * Cg(Cc(false), "is_opening") 
+                                  * Cg(Cc(false), "is_opening")
                                   * ( Cg(Cc(true), "is_direct")
                                     * Cg(Ct(C(parsers.rbracket) * Ct(parsers.inline_direct_ref)), "content")
                                     + Cg(Cc(false), "is_direct")
                                     * Cg(parsers.rbracket, "content")))
 
-  parsers.link_image_open_or_close  = parsers.link_image_opening 
+  parsers.link_image_open_or_close  = parsers.link_image_opening
                                     + parsers.link_image_closing
 
   parsers.link_image_content  = Ct( Cg(Cc("content"), "type")
-                              * Cg(C(((parsers.backslash^-1 / "" * parsers.any - parsers.link_image_open_or_close))^0), "content"))
+                                  * Cg(C(((parsers.backslash^-1 / "" * parsers.any - parsers.link_image_open_or_close))^0), "content"))
 
   parsers.link_array = C((parsers.exclamation^-1 * parsers.any_link)^1)
 
   parsers.link_group = Ct( parsers.link_image_opening
                         * (parsers.link_image_content * parsers.link_image_open_or_close)^0)
 
+  -- return the dangling parentheses of an inline link
+  -- exactly as they were in the input
   local function resolve_inline_link_closer(t, index, include_bracket)
```

Comment:
This is a definite improvement over not having any explanation, although I would still appreciate full-blown prose that can be typeset in the technical documentation and that allows you to describe the function in detail. For example, the following is preferable (here and everywhere):

```suggestion
%    \end{macrocode}
% \begin{markdown}
%
% Return the dangling parentheses of an inline link exactly as they were in the input.
%
% \end{markdown}
%  \begin{macrocode}
```

Keeping the description grounded in the actual parameters of the function is also appreciated. For example, you talk about "input" in your description, but the function accepts `t`, `index`, and `include_brackets` as parameters. The desciption should describe the intent of the function but also help the reader to understand what the parameters are.